keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigController.kt

```diff
@@ -75,19 +79,23 @@ class DeliveryConfigController(
     produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
   )
   @PreAuthorize("@authorizationSupport.hasApplicationPermission('READ', 'APPLICATION', #deliveryConfig.application)")
-  fun diff(@RequestBody deliveryConfig: SubmittedDeliveryConfig): List<EnvironmentDiff> =
-    adHocDiffer.calculate(deliveryConfig)
+  fun diff(@RequestBody deliveryConfig: SubmittedDeliveryConfig): List<EnvironmentDiff> {
+    validator.validate(deliveryConfig)
+    return adHocDiffer.calculate(deliveryConfig)
+  }
 
   @PostMapping(
     path = ["/validate"],
     consumes = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
     produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
   )
   @PreAuthorize("@authorizationSupport.hasApplicationPermission('READ', 'APPLICATION', #deliveryConfig.application)")
-  fun validate(@RequestBody deliveryConfig: SubmittedDeliveryConfig) =
-  // TODO: replace with JSON schema/OpenAPI spec validation when ready (for now, leveraging parsing error handling
+  fun validate(@RequestBody deliveryConfig: SubmittedDeliveryConfig) {
+    // TODO: replace with JSON schema/OpenAPI spec validation when ready (for now, leveraging parsing error handling
     //  in [ExceptionHandler])
```

Comment:
Honestly, I think we can remove this `TODO` at this point. If we eventually use the OpenAPI spec here, that's fine, but this is our current validation framework and it's fine.

Reply:
I agree, removing those.